File: study_for_me/study_manager/lib/models/language_model.dart
class Language {
  final String? id;
  final String name;
  final String description;
  final double progress;

  Language({this.id, required this.name, required this.description, this.progress = 0});

  factory Language.fromJson(Map<String, dynamic> json) {
    return Language(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      progress: json['progress']?.toDouble() ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'progress': progress,
    };
  }
}

File: study_for_me/study_for_me/study_manager_backend/src/models/language_model.js
const mongoose = require('mongoose');

const LanguageSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: { type: String, required: true },
  progress: { type: Number, default: 0 }, // Progresso de 0 a 100%
});

module.exports = mongoose.model('Language', LanguageSchema);